manage-docker.sh

```diff
@@ -1,73 +1,90 @@
+        echo "    shell     attach to running container and run a shell"
+        echo "    port      query the port where the appserver is bound to"
+        echo "    logs      tail the logs of dropwizard"
+        echo "ProTip: source this file for tab completion"
+    else
 
-if [ $(whoami) != "root" ]; then echo "[i] this script has to be executed as root!" && exit 1; fi
+        set -e # exit on error
 
-function port() {
-    docker port "${CONTAINER}"
-}
-function build() {
-    docker build -t "${IMAGE}" .
-}
-function run() {
-    docker run -d -p 80 -p 8081 --volume="${VOL_LOC}:${VOL_DOC}" \
-        --name="${CONTAINER}" "${IMAGE}"
-}
-function up() {
-    if [ ! -e "${CONF_DIR}/config.yml" ] && [ ! -e "${CONF_DIR}/minimumVersion.yml" ]; then
-        echo "[x] ${CONF_DIR}/config.yml or ${CONF_DIR}/minimumVersion.yml is missing"
-        exit 1
-    else
-        build
-        run
-    fi
-}
-function start() {
-    docker start "${CONTAINER}"
-}
-function stop() {
-    docker stop "${CONTAINER}"
-}
-function restart() {
-    docker restart "${CONTAINER}"
-}
-function clean() {
-    read -p "Do you really want to delete the container including ALL data? [y/N] "
-    if [ "$REPLY" == "y" ]; then
-        docker rm -v "${CONTAINER}"
-    fi
-}
-function shell() {
-    docker exec -it "${CONTAINER}" "${SHELL}"
-}
-function logs() {
-    docker logs -f "${CONTAINER}"
-}
+        IMAGE="app-server"
+        CONTAINER="${IMAGE}"
+        CONF_DIR="${PWD}/conf/"
+        VOL_LOC="${CONF_DIR}"
+        VOL_DOC="/home/fablab/app-server/src/dist/"
+        SHELL="bash"
 
-FUNCTIONS=("up" "run" "start" "stop" "restart" "build" "clean" "shell" "port" "logs")
+        function port() {
+            docker port "${CONTAINER}"
+        }
+        function build() {
+            docker build -t "${IMAGE}" .
+        }
+        function run() {
+            docker run -d -p 80 -p 8081 --volume="${VOL_LOC}:${VOL_DOC}" \
+                --name="${CONTAINER}" "${IMAGE}"
+        }
+        function run-shell() {
+            docker run --rm -it -p 80 -p 8081 --volume="${VOL_LOC}/:${VOL_DOC}/" \
+                --name="${CONTAINER}-interactive" "${IMAGE}" "${SHELL}"
+        }
+        function up() {
+            if [ ! -e "${CONF_DIR}/config.yml" ] && [ ! -e "${CONF_DIR}/minimumVersion.yml" ]; then
+                echo "[x] ${CONF_DIR}/config.yml or ${CONF_DIR}/minimumVersion.yml is missing"
+                exit 1
+            else
+                build
+                run
+            fi
+        }
+        function start() {
+            docker start "${CONTAINER}"
+        }
+        function stop() {
+            docker stop "${CONTAINER}"
+        }
+        function restart() {
+            docker restart "${CONTAINER}"
+        }
+        function clean() {
+            read -p "Do you really want to delete the container including ALL data? [Y/n] "
+            if [ "$REPLY" == "y" ] || [ -z $REPLY ]; then
```

Comment:
I think we shouldn't change the default behavior to `yes`.
It is a difference whether to remove only the docker container or to remove it including all associated volumes.
In our case all associated volumes will be removed with it (notice the `-v` flag in `docker rm`) and this might not be the users intention when using a `clean` command.